fhir/resources/core/fhirabstractmodel.py

```diff
@@ -177,10 +179,10 @@ def add_root_validator(
 
     @classmethod
     def element_properties(
-        cls: typing.Type["Model"],
-    ) -> typing.Generator[ModelField, None, None]:
+        cls: typing.Type["BaseModel"],
+    ) -> typing.Generator[FieldInfo, None, None]:
         """ """
-        for model_field in cls.__fields__.values():
+        for model_field in cls.model_fields:
```

Comment:
`ModelField` was removed in pydantic v2. not sure this is the correct replacement yet though